### ISSUE TITLE
Zed Editor Support

### ISSUE THREAD

**Issue by HarshNarayanJha** [paul-gauthier/aider#851]
### Issue

Aider seems to be a nice assistant, and it even works with Google Gemini! It would be nice if aider can be integrated into the Zed Editor, via some kind of extension.

https://github.com/zed-industries/zed

### Version and model info

Not Relevant.

**Comment by transcendr** [comment]
Aider is a command line application...

**Comment by HarshNarayanJha** [comment]
I was talking about something similar to this: https://marketplace.visualstudio.com/items?itemName=MattFlower.aider

> https://aider.chat/docs/install/optional.html#other-editors

**Comment by Chewt** [comment]
Those plugins are made by other people, not the aider dev. You could try writing a plugin yourself, using aider to help you do it, which is what the neovim plugin author did ;)

**Comment by HarshNarayanJha** [comment]
Zed doesn't support extensions other than language support or themes yet.

**Comment by torniker** [comment]
You could run: 
```
aider --watch-files
```

and communicate with comments.

https://aider.chat/docs/usage/watch.html

**Comment by KarmaMonk** [comment]
Zed publised a new Agent Client Protocol (ACP), which allows to integrate an AI Agent into an IDE.
https://agentclientprotocol.com/overview/introduction

It would be nice if Aider would support this new protocol,

**Comment by PsiACE** [comment]
Hi! It's great to see you discussing the possibility of supporting acp or zed. I've recently spent some time updating the Python SDK to make it more developer-friendly, and at least some users are already leveraging it to build coding agents. If you're open to it, I can explore whether I might be able to offer some assistance.

Check out https://github.com/PsiACE/agent-client-protocol-python

**Comment by guitcastro** [comment]
I would suggest rename the issue to let aider have ACP support. This would help develop aider plugins for the entire ecosystem, not just zed.

**Comment by KeithCu** [comment]
Yes, this bug should be renamed. ACP support is a great feature that lets people talk to agents without needing to use the "ugly" command line. I've built a little LibreOffice plugin (https://github.com/KeithCu/writeragent) that has a custom simple agent inside, but it also supports MCP/ACP which lets it talk to an agent without TTY. Please consider adding this feature to let people easily use your agent in more apps.

I'm going to look into making a PR about it, if I can, but if not someone else should since it's pretty easy I believe. Hermes-Agent added it recently. I tried talking to the agents via TTY and it was messy and unreliable and it would often hang for some reason.

Update: I copied over the code from Hermes-Agent (MIT) and adapted it in and submitted a PR. I need to implement the other side in my plugin to be sure it works, so I copied over the Hermes-Agent test code and put that in the commit. If people want to try it out the PR is here: https://github.com/Aider-AI/aider/pull/4936

**Comment by Displacer** [comment]
Is there any update on the PR review? While we are happy with aider as a CLI tool, ACP support is still needed for those who want to use a GUI. Since even Qt Creator is expected to support ACP (prepared in the 20.0 beta), many people might consider using OpenCode instead of aider for this

**Comment by EnteRth3Void** [comment]
If it’s useful while native ACP ([#4936](https://github.com/Aider-AI/aider/pull/4936)) is still open: there’s a preview community adapter that runs Aider as a Zed External Agent over ACP.

This is not a wrapper around the aider CLI. It imports Aider’s Python coder and talks ACP on stdin/stdout, so there is no PTY scraping or subprocess parsing.

Edits go through Zed’s review multibuffer (accept/reject per file) instead of writing straight to disk. Keys come from env (AIDER_ACP_MODELS + provider keys). Not affiliated with Aider-AI.

https://github.com/EnteRth3Void/aider_acp

Works in Zed’s Agent Panel: slash commands, @ to add files, stop/cancel, token usage, and per-file accept/reject review.

Not in this preview: loading old threads, Aider’s git auto-commit//undo flow, and an in-UI login (API keys stay in agent_servers.env).

**Comment by KeithCu** [comment]
Has anyone tried out my PR? I forgot about it, but I'd be happy to dig into it and fix issues if there are any. I don't use Aider currently so I never got around to testing it. In the meanwhile, anyone **else** can just make a build with this PR and report if it works for them: https://github.com/Aider-AI/aider/pull/4936

**Comment by yun520-1** [comment]
The `--watch-files` workaround is useful for the specific case of file-based communication, but it does not address what Zed integration actually needs, which is **shared editor state**.

The deeper question this issue raises is: what does an AI coding agent look like when it is native to an editor, rather than visiting via terminal? Aider's current model is process-per-session with file-based I/O. Zed's extension model expects long-lived connections with structured events (open, edit, save, close) and bidirectional streaming. Bridging them requires more than `aider --watch-files` — it requires an event adapter that translates Zed's edit events into aider's file-watch inputs and translates aider's proposed edits into Zed's apply-edit protocol.

The interesting part from an agent architecture perspective: aider already has the right primitive (watch mode), but it is framed as a CLI flag, not as a transport layer. If the watch input were a socket or stdio protocol instead of filesystem polling, any editor with an extension API could drive aider without aider needing editor-specific code. This is the same shape as the MCP server pattern — a tool that speaks a protocol, not a product that speaks an integration.
